include/matx/operators/interp.h

```diff
@@ -428,105 +428,46 @@ namespace matx {
         // Allocate temporary storage for spline coefficients
         if (method_ == InterpMethod::SPLINE) {
           static_assert(is_cuda_executor_v<Executor>, "cubic spline interpolation only supports the CUDA executor currently");
+          cudaStream_t stream = ex.getStream();
 
-          index_t _batch_count = 1;
+          index_t batch_count = 1;
           for (int i = 0; i < v_.Rank() - 1; i++) {
-            _batch_count *= v_.Size(i);
-          }
-          index_t _n = v_.Size(v_.Rank() - 1);
-          if (_batch_count > std::numeric_limits<int>::max() || _n > std::numeric_limits<int>::max()) {
-            const std::string err_msg = "Spline interpolation is not supported for tensors with more than 2^" + std::to_string(std::numeric_limits<int>::digits) + " items";
-            MATX_THROW(matxInvalidSize, err_msg.c_str());
+            batch_count *= v_.Size(i);
           }
-          int batch_count = static_cast<int>(_batch_count);
-          int n = static_cast<int>(_n); // number of sample points
+          index_t n = v_.Size(v_.Rank() - 1);
 
+          
           cuda::std::array m_shape = v_.Shape();
           detail::AllocateTempTensor(m_, std::forward<Executor>(ex), m_shape, &ptr_m_);
 
-          detail::tensor_impl_t<value_type, OpV::Rank()> d_tensor, dl_tensor, du_tensor; // Derivatives at sample points (spline only)
-          value_type *ptr_dl_ = nullptr;
-          value_type *ptr_d_ = nullptr;
-          value_type *ptr_du_ = nullptr;
+          // Allocate temporary storage for tridiagonal system
+          // use a single buffer for all three diagonals so that we can use the DIA format
+          value_type *ptr_tridiag_ = nullptr;
+          matxAlloc((void**)&ptr_tridiag_, 3 * batch_count * n * sizeof(value_type), MATX_ASYNC_DEVICE_MEMORY, stream);
```

Comment:
This currently causes a segfault, as it creates the offsets in the same memory space (and then tries to access them later from the host)